src/services/TxHistory/helpers.tsx

```diff
@@ -53,13 +49,20 @@ export const makeTxReceipt = (
 };
 
 export const merge = (apiTxs: ITxReceipt[], accountTxs: ITxReceipt[]): ITxReceipt[] => {
-  // Prioritize Account TX - needs to be more advanced?
-  const filteredApiTxs = apiTxs.filter(
-    (tx) => !accountTxs.find((a) => isSameHash(a.hash, tx.hash))
-  );
-  return filteredApiTxs.concat(accountTxs);
+  const apiTxsHashMap = convertTxsToHashMap(apiTxs);
+  const accountTxsHashMap = convertTxsToHashMap(accountTxs, apiTxsHashMap);
+  return Object.values(accountTxsHashMap);
 };
 
+export const convertTxsToHashMap = (txs: ITxReceipt[], initialMap?: Record<string, ITxReceipt>) =>
+  txs.reduce<Record<string, ITxReceipt>>(
+    (acc, cur) => ({
+      ...acc,
+      [cur.hash.toLowerCase()]: initialMap ? { ...acc[cur.hash.toLowerCase()], ...cur } : cur
```

Comment:
Is it not possible for the hash to not be in `initialMap`?

Reply:
It is possible, but when that occurs it shouldn't change the output of the object, correct?